contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5193,3 +5197,32 @@ unique_constraint_nulls_ordering(ConstrType constraint_type, SortByDir ordering)
 
 	return SORTBY_NULLS_DEFAULT;
 }
+
+/*
+ * is_partitioned_table_reloptions_allowed
+ * 	This function checks if the given reloptions are allowed or not for partitioned tables.
+ * 	Returns true if allowed, false otherwise.
+ * 
+ * 	Only bbf_rel_create_date and bbf_original_rel_name reloptions
+ * 	are allowed in TSQL dialect and while restoring babelfish database.
+ */
+static bool
+is_partitioned_table_reloptions_allowed(Datum reloptions)
```

Comment:
This hook seems to be in a hot path in Native PG, lets discuss if we can do it in another way!